src/commands/deploy.js

```diff
@@ -16,20 +17,265 @@ const SitesCreateCommand = require('./sites/create')
+  if (flags['dir']) {
+    deployFolder = path.resolve(process.cwd(), flags['dir'])
+  } else if (get(config, 'build.publish')) {
+    deployFolder = path.resolve(site.root, get(config, 'build.publish'))
+  } else if (get(siteData, 'build_settings.dir')) {
+    deployFolder = path.resolve(site.root, get(siteData, 'build_settings.dir'))
+  }
+
+  if (!deployFolder) {
+    log('Please provide a publish directory (e.g. "public" or "dist" or "."):')
+    log(process.cwd())
+    const { promptPath } = await inquirer.prompt([
+      {
+        type: 'input',
+        name: 'promptPath',
+        message: 'Publish directory',
+        default: '.',
+        filter: input => path.resolve(process.cwd(), input),
+      },
+    ])
+    deployFolder = promptPath
+  }
+
+  return deployFolder
+}
+
+const validateDeployFolder = async ({ deployFolder, error }) => {
+  let stat
+  try {
+    stat = await statAsync(deployFolder)
+  } catch (e) {
+    if (e.code === 'ENOENT') {
+      return error(`No such directory ${deployFolder}! Did you forget to run a build?`)
+    }
+
+    // Improve the message of permission errors
+    if (e.code === 'EACCES') {
+      return error('Permission error when trying to access deploy folder')
+    }
+    throw e
+  }
+
+  if (!stat.isDirectory()) {
+    return error('Deploy target must be a path to a directory')
+  }
+  return stat
+}
+
+const getFunctionsFolder = ({ flags, config, site, siteData }) => {
+  let functionsFolder
+  // Support "functions" and "Functions"
+  const funcConfig = get(config, 'build.functions') || get(config, 'build.Functions')
+  if (flags['functions']) {
+    functionsFolder = path.resolve(process.cwd(), flags['functions'])
+  } else if (funcConfig) {
+    functionsFolder = path.resolve(site.root, funcConfig)
+  } else if (get(siteData, 'build_settings.functions_dir')) {
+    functionsFolder = path.resolve(site.root, get(siteData, 'build_settings.functions_dir'))
+  }
+  return functionsFolder
+}
+
+const validateFunctionsFolder = async ({ functionsFolder, log, error }) => {
+  let stat
+  if (functionsFolder) {
+    // we used to hard error if functions folder is specified but doesn't exist
+    // but this was too strict for onboarding. we can just log a warning.
+    try {
+      stat = await statAsync(functionsFolder)
+    } catch (e) {
+      if (e.code === 'ENOENT') {
+        log(
+          `Functions folder "${functionsFolder}" specified but it doesn't exist! Will proceed without deploying functions`
+        )
+      }
+      // Improve the message of permission errors
+      if (e.code === 'EACCES') {
+        error('Permission error when trying to access functions folder')
+      }
+    }
+  }
+
+  if (stat && !stat.isDirectory()) {
+    error('Functions folder must be a path to a directory')
+  }
+
+  return stat
+}
+
+const validateFolders = async ({ deployFolder, functionsFolder, error, log }) => {
+  const deployFolderStat = await validateDeployFolder({ deployFolder, error })
```

Comment:
Can we run those two `async` functions in parallel (`Promise.all()`)?

Reply:
`validateFunctionsFolder` prints a warning message when the folder doesn't exist, while `validateDeployFolder` exists.
Didn't want to run into a case where we print a warning then exit with another error.

Reply:
Makes sense :+1: 
This is micro-optimization anyway since those I/O calls are quite fast.